core/wiki/macros/toc.tid

```diff
@@ -1,218 +1,422 @@
 title: $:/core/macros/toc
 tags: $:/tags/Macro
 
+<!--
+Naming rules / pattern
+- functions: tf.toc-variableName, "tf." -> function prefix reservered for TW, "toc-" -> global macro prefix, "variableName"
+- procedures eg: toc-procedureName,
+- macro parameters: path, itemTemplate
+- derived macro parameters: _path, _itemTemplate -> derived variables start with an underscore
 
-\define toc-open-icon() $:/core/images/down-arrow
-\define toc-closed-icon() $:/core/images/right-arrow
+The toc macro code gets more and more complex from top to bottom.
+toc-tabbed-internal-nav calls toc-tabbed-external-nav, which callse toc-selective-expandable, which calls toc-expandable
+
+The very first toc macro is _not_ "reused". So it can be used as a "simple" reverence implementation,
+because it is realatively easy to debug.
+-->
 
-\define toc-caption()
 \whitespace trim
-<span class="tc-toc-caption tc-tiny-gap-left">
-<$set name="tv-wikilinks" value="no">
-  <$transclude field="caption">
-    <$view field="title"/>
-  </$transclude>
-</$set>
+
+\procedure toc-defaultItemTemplate()
+<span class="tc-toc-caption tc-tiny-gap-left" title={{!!tooltip}}>
+<$let tv-wikilinks="no">
+	<$transclude $tiddler={{{ [<itemTemplate>!is[blank]then<itemTemplate>else<currentTiddler>] }}}
+		$field={{{ [<itemTemplate>!is[blank]then[text]] :else[<captionField>!is[blank]then<captionField>else[caption]] }}}
+	>
+		<$view field="title"/>
+	</$transclude>
+</$let>
 </span>
 \end
 
-\define toc-body(tag,sort:"",itemClassFilter,exclude,path)
-\whitespace trim
+<!-- returns the title of a tiddler -->
+\function tf.toc-getItemTemplate() [<itemTemplate>!is[blank]then<itemTemplate>] :else[<currentTiddler>get[toc-itemTemplate]]
+
+<!-- Use macro itemTemplate or one defined as a toc-itemTemplate field -->
+\procedure _itemTemplate()
+<$transclude $tiddler=<<tf.toc-getItemTemplate>> >
+	<<toc-defaultItemTemplate>>
+</$transclude>
+\end
+
+<!-- v5.2.4 add "Custom Icons" -->
```

Comment:
What does this comment mean? Is it just marking something that was added in v5.2.4? Why is it needed?

Reply:
I did add this info to see, that I cannot change those names for backwards compatibility reasons. Otherwise I may have changed them by accident

I'll change it to: `<!-- Allow custom icons (TW v5.2.4). Keep names for backwards compatibility -->`